csrc/host_ir/lower_to_llvm.cpp

```diff
@@ -0,0 +1,704 @@
+  auto* entry = llvm::BasicBlock::Create(*ctx, "entry", func);
+  builder.SetInsertPoint(entry);
+  std::unordered_map<Val*, dependency_graph> val2graph;
+  std::vector<Val*> input_vals = domain2vals(input_logical_domain);
+  std::vector<Val*> output_vals = domain2vals(output_logical_domain);
+  auto arg_it = func->arg_begin();
+  llvm::Value* input_ptr = &*arg_it;
+  llvm::Value* output_ptr = &*arg_it+2;
+  IdModel id_model(&fusion);
+  const ValGraph& graph = id_model.buildExactGraph();
+  auto exprs = traverse_expr_group(graph, input_logical_domain, output_logical_domain);
+  for(long unsigned int i = 0; i < input_logical_domain.size(); i++){
+    auto* zero = builder.getInt64(i);
+    auto* input_i_ptr = builder.CreateGEP(int64Ty, input_ptr, zero, "ptr");
+    auto* input_i_val = builder.CreateLoad(int64Ty, input_i_ptr, "val");
+    val2graph[input_vals[i]] = dependency_graph();
+    val2graph[input_vals[i]].llvm_val = input_i_val;
+  }
+
+  // Map the output values to the input values if they are the same
+  for(auto* val : output_vals){
+    auto index = mapToInputDomain(input_vals, val, graph);
+    if(index != -1){
+      val2graph[val] = val2graph[input_vals[index]];
+    }
+  }
+
+  for(auto* expr : exprs){
+    auto expr_input_domain = vals2domain(expr->inputs());
+    build_dep_graph(expr, val2graph, builder);
+    int current_index = 0;
+    for(auto* expr_input : expr->inputs()){
+      auto index = mapToInputDomain(input_vals, expr_input, graph);
+      if(index != -1){
+        val2graph[expr_input].input_vals[current_index] = input_vals[index];
+      }
+      current_index++;
+    }
+  }
+
+  for(auto* val : output_vals){
+    generate_shape_llvm_ir(val, val2graph, builder);
+  }
+
+  for(long unsigned int i = 0; i < output_vals.size(); i++){
+    auto* output_i_ptr = builder.CreateGEP(int64Ty, output_ptr, builder.getInt64(i), "ptr");
+    builder.CreateStore(val2graph[output_vals[i]].llvm_val, output_i_ptr);
+  }
+
+  builder.CreateRetVoid();
+  llvm::outs() << "=== LLVM IR ===\n";
+  Module->print(llvm::outs(), nullptr);
+  return llvm::orc::ThreadSafeModule(std::move(Module), std::move(Context));
+}
+
+// JIT add module wrapper function
+std::unique_ptr<llvm::orc::LLJIT> llvm_jit_init(int num_threads){
+  llvm::InitializeNativeTarget();
+  llvm::InitializeNativeTargetAsmPrinter();
+  auto JIT = ExitOnErr(llvm::orc::LLJITBuilder().setNumCompileThreads(num_threads).create());
+  return JIT;
+}
+
+// shape infer compile
+void llvm_jit_compile_shape_infer(std::unique_ptr<llvm::orc::LLJIT>& JIT, Fusion& fusion, std::vector<IterDomain*>& input_domain, std::vector<IterDomain*>& output_domain){
+  std::cout << "llvm_jit_compile shape infer" << std::endl;
+  auto TSM_shape = generate_infer_shape_module(input_domain, output_domain, fusion);
+  if (auto Err = JIT->addIRModule(std::move(TSM_shape))) {
+    llvm::errs() << "Error adding module to JIT: " << llvm::toString(std::move(Err)) << "\n";
```

Comment:
nitpick: do we need to use `llvm::errs()`? There's NVF_ERROR for that.
https://github.com/NVIDIA/Fuser/blob/main/csrc/exceptions.h